dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -217,6 +219,12 @@ public Response toolsIndexGet(SecurityContext securityContext) {
 
     @Override
     public Response toolsIndexSearch(String query, MultivaluedMap<String, String> queryParameters, SecurityContext securityContext) {
+        String unableToUseESMsg = "Could not use Elasticsearch search";
+        if (!ELASTIC_SEARCH_CONCURRENCY_LIMIT.tryAcquire(1)) {
```

Comment:
I was looking at tryAcquire (hadn't used it before), and there's an overloaded method that takes a timeout parameter. You could wait a second or two before erroring out.

How long does an ES request typically take? If it's under a second, it might be worth waiting a second.

Reply:
So, if you're not spamming the search, the requests take less than a second. However, once you start spamming, the timing really starts ramping up after 20-30ish requests. Also, each search in the UI makes multiple requests to this endpoint.